src/VernaContextProvider.tsx

```diff
@@ -168,17 +168,24 @@ function VernaContextProvider({
     if (selector) {
       if (fullSchema['properties']) setSchema(fullSchema['properties'][selector] as JSONSchema7);
       if (fullUiSchema) setUiSchema(fullUiSchema[selector]);
-      if (defaultFormValues) setDefaultValues(defaultFormValues[selector]);
     } else {
       _setSchema(fullSchema);
       _setUiSchema(fullUiSchema);
-      setDefaultValues(defaultFormValues);
     }
   }, [selector]);
 
-  function getFormData(localFormData: unknown) {
+  function getFormDataToSubmit(localFormData: unknown) {
     if (!selector) return localFormData;
-    else return { ...defaultFormValues, [selector]: localFormData };
+    else {
+      const newFormData = { ...formData, [selector]: localFormData };
+      console.log('setformdata:', newFormData);
+      setFormData(newFormData);
+      return newFormData;
+    }
+  }
+
+  function getSelectedFormData() {
+    return selector && formData ? formData.properties[selector] : formData;
   }
```

Comment:
Instead a function we could memoized the selected form data.

```suggestion
const selectedFormData = useMemo(() => {
    return selector && formData ? formData.properties[selector] : formData;
  }, [formData, selector])
```

Reply:
I never think about doing it that way !